### PR TITLE
Add Prometheus endpoint

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
Metrics

### Description
This PR adds a `/metrics` Prometheus endpoint using https://github.com/PayU/prometheus-api-metrics.
The custom stats currently available at `/status` are also available as gauges.

I quickly build this for my internal monitoring, but thought it might be nice to have upstream as well.
I never created any Prometheus metrics before, so feedback on the naming/structure is welcome!

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

**Comment by SISheogorath** [comment]
I think it would be useful to move the Prometheus client definition out if the utils Funktion as I don't think it's useful to always have realtime in that context.

File: lib/utils.js [hunk 1]
```diff
@@ -25,3 +27,47 @@ exports.getImageMimeType = function getImageMimeType (imagePath) {
       return undefined
   }
 }
+
+exports.setupCustomPrometheusMetrics = function () {
```
Comment (DerMolly) [inline lib/utils.js:31]: I'd suggest moving this in its own prometheus.ts file

**Review by InnayTool — DISMISSED** [review]

**Comment by davidmehren** [comment]
The function `setupCustomPrometheusMetrics` has been moved to its own `metrics.js` file.

**Comment by davidmehren** [comment]
As suggested by @ErikMichelson, the new endpoint is now documented.

**Review by DerMolly — APPROVED** [review]

**Review by InnayTool — DISMISSED** [review]

File: lib/metrics.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    name: 'hedgedoc_online_sessions',
+    help: 'Sessions currently editing notes',
+    labelNames: ['type']
+  })
+  const onlineUsers = new promClient.Gauge({
+    name: 'hedgedoc_online_users',
+    help: 'Distinct users currently editing notes',
+    labelNames: ['type']
+  })
+  const notesCount = new promClient.Gauge({ name: 'hedgedoc_notes', help: 'Notes in the instance' })
```
Comment (ErikMichelson) [inline lib/metrics.js:19]: Any reason why this Gauge's config is inlined and not like the other ones split into multiple lines?

Comment (davidmehren) [reply lib/metrics.js:19]: I think the line is short enough to not make eslint unhappy

**Review by ErikMichelson — DISMISSED** [review]

**Review by InnayTool — DISMISSED** [review]

**Review by DerMolly — DISMISSED** [review]

**Review by DerMolly — APPROVED** [review]

**Review by InnayTool — DISMISSED** [review]

**Comment by SISheogorath** [comment]
The library file should either be called `prometheus.js` or also contain the code from the `/status` endpoint. I don't think it makes a lot of sense to spread this code around like this when using more universal names.

**Comment by davidmehren** [comment]
I opted for renaming to `prometheus.js`, moving the `/status` code didn't seem to make much sense, as it's mostly an Express router.

**Review by DerMolly — APPROVED** [review]

**Review by SISheogorath — APPROVED** [review]

**Pull request merged**